fetchlogs

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python
+
+import argparse
+import exceptions
+import os
+import shutil
+import subprocess
+import sys
+import zipfile
+
+
+def fetch_logs():
+    containers_with_logfiles = {
+        'bloc': '/logs',
+        'strato': '/var/lib/strato/logs'
+    }
+    try:
+        running_containers = subprocess.check_output(
+            "docker-compose --log-level ERROR -p strato ps | grep strato_ | awk '{print $1}'",
+            shell=True
+        ).split('\n')
+        running_containers = filter(None, running_containers)
+    except exceptions.Exception as e:
+        print('unable to fetch the currently running containers:\n%s' % e)
+        sys.exit(6)
+
+    for container_name in running_containers:
+        container_name_split = container_name.split('_')
+        if len(container_name_split) != 3:
+            print('unexpected container name: %s' % container_name)
+            sys.exit(7)
+        container_subname = container_name_split[1]
+        if container_subname in containers_with_logfiles.keys():
+            current_container_logs_dir = os.path.join(temp_dir_path, container_subname)
+            os.makedirs(current_container_logs_dir)
+            docker_cp_command = 'docker cp %s:%s %s' % (
+                container_name,
+                os.path.join(containers_with_logfiles[container_subname], '.'),
+                current_container_logs_dir
+            )
+            try:
+                subprocess.check_output(docker_cp_command, shell=True)
+            except exceptions.Exception as e:
+                error_message = 'unable to fetch logs from %s container with error:\n%s' % (container_name, e)
+                print(error_message)
+                create_text_file(os.path.join(current_container_logs_dir, '%s_error.txt' % container_subname),
+                                 error_message)
+        else:
+            # Copying logfile returned with `"docker inspect --format='{{.LogPath}}' %s" % container_name` does not work
+            # since: 1. it returns non-existing log filenames in some cases 2. sudo is mandatory to access file directly
+            try:
+                subprocess.check_output(
```

Comment:
I see a syntax error from `sh` when I try this:
```
Fetching STRATO logs - this may take up to a minute...
/bin/sh: 1: Syntax error: Bad fd number
unable to fetch logs of strato_apex_1 container with error:
Command 'docker logs strato_apex_1 >& ./strato_logs/apex' returned non-zero exit status 2
/bin/sh: 1: Syntax error: Bad fd number
unable to fetch logs of strato_dappstore_1 container with error:
Command 'docker logs strato_dappstore_1 >& ./strato_logs/dappstore' returned non-zero exit status 2
/bin/sh: 1: Syntax error: Bad fd number
unable to fetch logs of strato_docs_1 container with error:
Command 'docker logs strato_docs_1 >& ./strato_logs/docs' returned non-zero exit status 2
/bin/sh: 1: Syntax error: Bad fd number
unable to fetch logs of strato_kafka_1 container with error:
Command 'docker logs strato_kafka_1 >& ./strato_logs/kafka' returned non-zero exit status 2
/bin/sh: 1: Syntax error: Bad fd number
unable to fetch logs of strato_nginx_1 container with error:
Command 'docker logs strato_nginx_1 >& ./strato_logs/nginx' returned non-zero exit status 2
/bin/sh: 1: Syntax error: Bad fd number
unable to fetch logs of strato_postgres_1 container with error:
Command 'docker logs strato_postgres_1 >& ./strato_logs/postgres' returned non-zero exit status 2
/bin/sh: 1: Syntax error: Bad fd number
unable to fetch logs of strato_postgrest_1 container with error:
Command 'docker logs strato_postgrest_1 >& ./strato_logs/postgrest' returned non-zero exit status 2
/bin/sh: 1: Syntax error: Bad fd number
unable to fetch logs of strato_prometheus_1 container with error:
Command 'docker logs strato_prometheus_1 >& ./strato_logs/prometheus' returned non-zero exit status 2
/bin/sh: 1: Syntax error: Bad fd number
unable to fetch logs of strato_redis_1 container with error:
Command 'docker logs strato_redis_1 >& ./strato_logs/redis' returned non-zero exit status 2
/bin/sh: 1: Syntax error: Bad fd number
unable to fetch logs of strato_smd_1 container with error:
Command 'docker logs strato_smd_1 >& ./strato_logs/smd' returned non-zero exit status 2
/bin/sh: 1: Syntax error: Bad fd number
unable to fetch logs of strato_vault-wrapper_1 container with error:
Command 'docker logs strato_vault-wrapper_1 >& ./strato_logs/vault-wrapper' returned non-zero exit status 2
/bin/sh: 1: Syntax error: Bad fd number
unable to fetch logs of strato_zookeeper_1 container with error:
Command 'docker logs strato_zookeeper_1 >& ./strato_logs/zookeeper' returned non-zero exit status 2
Zipping data - almost done...
Done.
```
Should this be `&>`?

Reply:
Also, I think if extracting container logs fails that should be an exception from `fetchlogs`: We know that the container should exist from the docker-compose, so it should always have a response to `docker logs`

Reply:
Yes, I think it should be `&>` (somehow `>&` didn't fail for me on mac).

Regarding the exception from fetchLogs - the error in fetching one container's logs should not prevent fetching the other logs, so adding the `deferred_errors` list to show error messages in the end and exit script with non-zero code